components/scream/src/dynamics/homme/tests/homme_pd_remap_tests.cpp

```diff
@@ -268,35 +252,28 @@ TEST_CASE("remap", "") {
             h_ss_3d_view(idof,il) = gid;
             h_vs_3d_view(idof,0,il) = gid;
             h_vs_3d_view(idof,1,il) = gid;
-            for (int iq=0; iq<NTQ; ++iq) {
-              h_ts_3d_view(idof,iq,il) = gid;
+            for (int iq=0; iq<NQ; ++iq) {
+              h_tr_3d_view(idof,iq,il) = gid;
             }
           }
         }
-        Kokkos::deep_copy(s_2d_view,h_s_2d_view);
-        Kokkos::deep_copy(v_2d_view,h_v_2d_view);
-        Kokkos::deep_copy(s_3d_view,h_s_3d_view);
-        Kokkos::deep_copy(v_3d_view,h_v_3d_view);
-
-        Kokkos::deep_copy(ss_3d_view,h_ss_3d_view);
-        Kokkos::deep_copy(vs_3d_view,h_vs_3d_view);
-        Kokkos::deep_copy(ts_3d_view,h_ts_3d_view);
+        s_2d_field_phys.sync_to_dev();
+        v_2d_field_phys.sync_to_dev();
+        s_3d_field_phys.sync_to_dev();
+        v_3d_field_phys.sync_to_dev();
+
+        ss_3d_field_phys.sync_to_dev();
+        vs_3d_field_phys.sync_to_dev();
+        tr_3d_field_phys.sync_to_dev();
       } else {
-        auto s_2d_view = s_2d_field_dyn.get_reshaped_view<Homme::Real***>();
-        auto v_2d_view = v_2d_field_dyn.get_reshaped_view<Homme::Real****>();
-        auto s_3d_view = s_3d_field_dyn.get_reshaped_view<Homme::Real****>();
-        auto v_3d_view = v_3d_field_dyn.get_reshaped_view<Homme::Real*****>();
-        auto ss_3d_view = ss_3d_field_dyn.get_reshaped_view<Homme::Real*****>();
-        auto vs_3d_view = vs_3d_field_dyn.get_reshaped_view<Homme::Real******>();
-        auto ts_3d_view = ts_3d_field_dyn.get_reshaped_view<Homme::Real******>();
-
-        auto h_s_2d_view = Kokkos::create_mirror_view(s_2d_view);
-        auto h_v_2d_view = Kokkos::create_mirror_view(v_2d_view);
-        auto h_s_3d_view = Kokkos::create_mirror_view(s_3d_view);
-        auto h_v_3d_view = Kokkos::create_mirror_view(v_3d_view);
-        auto h_ss_3d_view = Kokkos::create_mirror_view(ss_3d_view);
-        auto h_vs_3d_view = Kokkos::create_mirror_view(vs_3d_view);
-        auto h_ts_3d_view = Kokkos::create_mirror_view(ts_3d_view);
+        auto h_s_2d_view = s_2d_field_dyn.get_reshaped_view<Homme::Real***,Host>();
+        auto h_v_2d_view = v_2d_field_dyn.get_reshaped_view<Homme::Real****,Host>();
+        auto h_s_3d_view = s_3d_field_dyn.get_reshaped_view<Homme::Real****,Host>();
+        auto h_v_3d_view = v_3d_field_dyn.get_reshaped_view<Homme::Real*****,Host>();
+        auto h_ss_3d_view = ss_3d_field_dyn.get_reshaped_view<Homme::Real*****,Host>();
+        auto h_vs_3d_view = vs_3d_field_dyn.get_reshaped_view<Homme::Real******,Host>();
+        auto h_tr_3d_view = tr_3d_field_dyn.get_reshaped_view<Homme::Real*****,Host>();
```

Comment:
Honest question: should there be one more `*` here (`Homme:Real******`)?

Reply:
Well, no. Before, we did have 6 `*`, cause I was considering for input/output the tracers mass, which is a 'state' in homme, so it has time levels: `Qdp(ie,itl,iq,igp,jgp,ilev)`. Now we consider for input/output the tracers concentration, which in homme is computed "diagnostically" from `Qdp` and `dp`, and has layout `Q(ie,iq,igp,jgp,ilev)`.